src/main/java/com/code_intelligence/jazzer/junit/FuzzTestExecutor.java

```diff
@@ -94,7 +95,16 @@ public static FuzzTestExecutor prepare(ExtensionContext context, String maxDurat
     ArrayList<String> libFuzzerArgs = new ArrayList<>();
     libFuzzerArgs.add(argv0);
 
-    // Store the generated corpus in a per-class directory under the project root, just like cifuzz:
+    // Add passed in corpus directories (and files) at the beginning of the arguments list,
+    // as those have precedence over the JUnit specific ones added later on.
+    List<String> corpusDirs = originalLibFuzzerArgs.stream()
```

Comment:
Instead of mutating the list further down, you could use [`groupingBy`](https://docs.oracle.com/javase/8/docs/api/java/util/stream/Collectors.html#groupingBy-java.util.function.Function-) to split the list into two parts.

Reply:
I tried that out originally, but didn't like the resulting code. Mutating the original list seemed simpler than handling the map returned by `groupingBy`.